card/lib/card/auth.rb

```diff
@@ -13,19 +13,32 @@ module Auth
     class << self
       # Authenticates a user by their login name and unencrypted password.
       def authenticate email, password
-        accounted = Auth[email]
-        return unless accounted && (account = accounted.account) &&
-                      account.active?
-        if Card.config.no_authentication ||
-           password_authenticated?(account, password.strip)
-          accounted.id
+        account = Auth[email]
+        case
+        when !account                                 then nil
+        when !account.active?                         then nil
+        when Card.config.no_authentication            then account
+        when password_valid?(account, password.strip) then account
         end
```

Comment:
agree.  I should note that both Auth[] and Auth.authenticate now return +*account cards (rather than ids of cards with accounts), which is part of why it's simpler. So in authenticating PK's email, it would formerly return PK's id but now returns the card PK+*account

If we use either in mods, we will need to change things accordingly.